gapir/client/client.go

```diff
@@ -17,80 +17,118 @@ package client
 import (
 	"context"
 	"fmt"
+	"math"
 	"sync"
 	"time"
 
 	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/app/auth"
 	"github.com/google/gapid/core/app/crash"
+	"github.com/google/gapid/core/app/crash/reporting"
 	"github.com/google/gapid/core/app/status"
 	"github.com/google/gapid/core/context/keys"
+	"github.com/google/gapid/core/data/id"
 	"github.com/google/gapid/core/event/task"
 	"github.com/google/gapid/core/log"
 	"github.com/google/gapid/core/os/device"
 	"github.com/google/gapid/core/os/device/bind"
+	"github.com/google/gapid/core/os/file"
 	"github.com/google/gapid/gapir"
+	replaysrv "github.com/google/gapid/gapir/replay_service"
+	"github.com/google/gapid/gapis/database"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 )
 
-type tyLaunchArgsKey string
-
 const (
 	// LaunchArgsKey is the bind device property key used to control the command
 	// line arguments when launching GAPIR. The property must be of type []string.
-	LaunchArgsKey     tyLaunchArgsKey = "gapir-launch-args"
-	connectTimeout                    = time.Second * 10
-	heartbeatInterval                 = time.Millisecond * 500
+	LaunchArgsKey = "gapir-launch-args"
+	// gapirAuthTokenMetaDataName is the key of the Context metadata pair that
+	// contains the authentication token. This token is common knowledge shared
+	// between GAPIR client (which is GAPIS) and GAPIR server (which is GAPIR
+	// device).
+	gapirAuthTokenMetaDataName = "gapir-auth-token"
+	// gRPCConnectTimeout is the time allowed to establish a gRPC connection.
+	gRPCConnectTimeout = time.Second * 10
+	// heartbeatInterval is the delay between heartbeat pings.
+	heartbeatInterval = time.Second * 2
 )
 
-type clientInfo struct {
-	device               bind.Device
-	arch                 device.Architecture
-	abi                  *device.ABI
-	deviceConnectionInfo deviceConnectionInfo
-	connection           gapir.Connection
-	bgConnection         *backgroundConnection
+// ReplayExecutor must be implemented by replay executors to handle some live
+// interactions with a running replay.
+type ReplayExecutor interface {
+	// HandlePostData handles the given post data message.
+	HandlePostData(context.Context, *gapir.PostData) error
+	// HandleNotification handles the given notification message.
+	HandleNotification(context.Context, *gapir.Notification) error
+	// HandleFinished is notified when the given replay is finished.
+	HandleFinished(context.Context, error) error
```

Comment:
We may want to add another function here for any case when replay cannot be finished due to an error e.g. Connection, crash etc.

Reply:
That sounds good, but it is a behavioral change, so I don't think it fits in this cleanup PR that tries hard to NOT change the package behavior/interface (beyond renaming).